docs/madr/decisions/026-mesh-tcp-route.md

```diff
@@ -0,0 +1,106 @@
+# `MeshTCPRoute`
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/6343
+
+## Context and Problem Statement
+
+Now that we have the [new policy matching resources](./005-policy-matching.md),
+[new traffic route designed](https://github.com/kumahq/kuma/issues/4743), and
+[basic api of `MeshHTTPRoute`](https://github.com/kumahq/kuma/issues/5470)
+implemented, it's time to introduce `MeshTCPRoute`.
+
+### Goals
+
+Policy should allow to
+* reroute TCP traffic to different versions of a service or even completely
+  different service;
+* split TCP traffic between services with different tags implementing 
+  A/B testing or canary deployments.
+
+### Non-goals
+
+* TCP traffic mirroring
+
+  It would be nice for the policy to allow to mirror TCP traffic. Unfortunately
+  it seems to be a non-trivial task as according to short research, Envoy
+  doesn't support this case out of the box. If there will be need for
+  this feature, it should be implemented in the future iterations.
+
+  Ref:
+  * https://github.com/envoyproxy/envoy/issues/18172
+
+* `MeshGatewayTCPRoute`
+
+  According to [Traffic Route MADR](./011-mesh-traffic-route.md), every 
```

Comment:
@michaelbeaumont @lahabana I think we verbally changed our mind about that right? The idea at this moment is to have MeshHTTPRoute work both for the built-in gateway and mesh. So I guess MeshTCPRoute should work the same way

Reply:
Isn't it what the second paragraph says?

Reply:
Yes, that was my intention, but if it's not clear I might try to clarify it a little bit. Wdyt @lobkovilya?

Reply:
I'm closing this conversation. If you @lobkovilya my explanation is not clear enough, feel free to open it again, and I'll do my best to improve it.

Reply:
Sorry, I didn't mean to point out that something is wrong in MADR :) I just wanted to say that MeshTCPRoute is not exceptional anymore, we want the same thing for MeshHTTPRoute